webapp/channels/src/components/post/post_options.tsx

```diff
@@ -65,15 +65,22 @@ const PostOptions = (props: Props): JSX.Element => {
     const [showDotMenu, setShowDotMenu] = useState(false);
     const [showActionsMenu, setShowActionsMenu] = useState(false);
 
+    const {emitShortcutReactToLastPostFrom} = props.actions;
+
+    const toggleEmojiPicker = useCallback(() => {
+        setShowEmojiPicker(!showEmojiPicker);
+        props.handleDropdownOpened!(!showEmojiPicker);
+    }, [props.handleDropdownOpened, showEmojiPicker]);
+
     useEffect(() => {
         const locationToUse = props.location === 'RHS_COMMENT' ? Locations.RHS_ROOT : props.location;
         if (props.isLastPost &&
             (props.shortcutReactToLastPostEmittedFrom === locationToUse) &&
                 props.isPostHeaderVisible) {
             toggleEmojiPicker();
-            props.actions.emitShortcutReactToLastPostFrom(Locations.NO_WHERE);
+            emitShortcutReactToLastPostFrom(Locations.NO_WHERE);
         }
-    }, [props.isLastPost, props.shortcutReactToLastPostEmittedFrom]);
+    }, [props.isLastPost, props.shortcutReactToLastPostEmittedFrom, emitShortcutReactToLastPostFrom, props.location, props.isPostHeaderVisible, toggleEmojiPicker]);
```

Comment:
The exhaustive dependencies don't always work well with effects. Take this change for example.

`toggleEmojiPicker` will change if we change `showEmojiPicker`. Therefore, this effect will run. If all the other props are "aligned", this will call `toggleEmojiPicker`, which will update the value of `showEmojiPicker`. If it wasn't for the line that sets the emitted from to nowhere, you would have a infinite loop.

Effects are to be considered as "side effects to something changing". In this particular case, we don't care if `toggleEmojiPicker`, since it is just the action we take if something happens, and we just want to make sure it is up to date when that something happens. The "something" we were looking for originally here is whether we become / are no longer the last post, and where the shortcut was emitted from.

Not sure if we should reconsider that dependency list, but for sure `toggleEmojiPicker` doesn't make sense here (and probably also `emitShortcutReactToLastPostFrom`).

Reply:
ok, understood. I verified the behavior and it continued working as expected. But you are right about those potentially causing infinite re-renders if it wasn't for the condition.